app/src/main/java/org/mifos/selfserviceapp/ui/fragments/HomeFragment.java

```diff
@@ -0,0 +1,311 @@
+
+    @BindView(R.id.loan_button)
+    Button loan_button;
+
+    @BindView(R.id.share_button)
+    Button share_button;
+
+    @BindView(R.id.transaction_button)
+    Button transaction_button;
+
+    @BindView(R.id.transfer_button)
+    Button transfer_button;
+
+    @BindView(R.id.payment_button)
+    Button payment_button;
+
+    private List<LoanAccount> loanAccounts;
+    private List<SavingAccount> savingAccounts;
+    private List<ShareAccount> shareAccounts;
+    private double totalLoans;
+    private long clientId;
+
+    public static HomeFragment newInstance(long clientId) {
+        HomeFragment fragment = new HomeFragment();
+        Bundle args = new Bundle();
+        args.putLong(Constants.CLIENT_ID, clientId);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+        loanAccounts = new ArrayList<>();
+        savingAccounts = new ArrayList<>();
+        shareAccounts = new ArrayList<>();
+        totalLoans = 0;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        rootView = inflater.inflate(R.layout.fragment_home, container, false);
+
+        ButterKnife.bind(this, rootView);
+        homePresenter.attachView(this);
+        homePresenter.loadClient();
+        homePresenter.loadInfo();
+        showProgress();
+
+        application_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((BaseActivity) getActivity()).replaceFragment(
+                        LoanApplicationFragment.newInstance(), false, R.id.container);
+            }
+        });
+
+        transaction_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((BaseActivity) getActivity()).replaceFragment(
+                        RecentTransactionsFragment.newInstance(clientId), false, R.id.container);
+            }
+        });
+
+        saving_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((BaseActivity) getActivity()).replaceFragment(
+                        ClientAccountsFragment.newInstance(clientId), false, R.id.container);
+            }
+        });
+
+        loan_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((BaseActivity) getActivity()).replaceFragment(
+                        ClientAccountsFragment.newInstance(clientId), false, R.id.container);
+            }
+        });
+
+        share_button.setOnClickListener(new View.OnClickListener() {
```

Comment:
Use the ButterKnife @onClick annotation instead of implementing the on click method.
for example https://github.com/openMF/self-service-app/blob/development/app/src/main/java/org/mifos/selfserviceapp/ui/fragments/LoanAccountsDetailFragment.java#L143

do the same for above.